src/disassembler.cpp

```diff
@@ -1,18 +1,101 @@
 #include "disassembler.hh"
-#include <list>
+#include <cinttypes>
+#include <cstdio>
 #include <ostream>
-#include <unordered_map>
+#include <sstream>
+#include "output.hh"
 
+using std::cerr;
+using std::ostringstream;
 using std::pair;
-using std::vector;
+
+#define STRING_NAME(x) \
+    case operation::x: \
+        return #x
+
+std::string disassembler::to_str(const instruction& i) noexcept {
+    using operation = instruction::operation;
+
+    switch (i.m_op) {
+        STRING_NAME(NOP);
+        STRING_NAME(ADD);
+        STRING_NAME(SUB);
+        STRING_NAME(MUL);
+        STRING_NAME(DIV);
+        STRING_NAME(MOD);
+        STRING_NAME(POP);
+        STRING_NAME(EXT);
+        STRING_NAME(INC);
+        STRING_NAME(DEC);
+        STRING_NAME(AND);
+        STRING_NAME(IOR);
+        STRING_NAME(XOR);
+        STRING_NAME(NOT);
+        STRING_NAME(GTC);
+        STRING_NAME(PTC);
+        STRING_NAME(GTI);
+        STRING_NAME(PTI);
+        STRING_NAME(IDX);
+        STRING_NAME(DUP);
+        STRING_NAME(RND);
+        STRING_NAME(EXP);
+        STRING_NAME(SWP);
+        STRING_NAME(GTM);
+        STRING_NAME(GDT);
+        STRING_NAME(DP2);
+        STRING_NAME(TKL);
+        STRING_NAME(TJN);
+        case operation::PSI: {
+            int24_t value = i.m_arg.number;
+            ostringstream result;
```

Comment:
This function should take the ostream as an argument, rather than having multiple `std::ostringstream` instances for a single operation